### PR TITLE
Update to latest packer version

### PR DESCRIPTION
**Pull request by moredure**

### CODE REVIEW

File: builder/upcloud/step_create_server.go [hunk 1]
```diff
@@ -15,10 +16,10 @@ type StepCreateServer struct {
 }
 
 // Run performs the actual step
-func (s *StepCreateServer) Run(state multistep.StateBag) multistep.StepAction {
+func (s *StepCreateServer) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
 	// Extract state
 	ui := state.Get("ui").(packer.Ui)
-	service := state.Get("service").(service.Service)
+	svc := state.Get("service").(service.Service)
```
Comment (Jalle19) [inline builder/upcloud/step_create_server.go:22]: Why change these variable names?

File: builder/upcloud/builder.go [hunk 2]
```diff
@@ -110,11 +117,6 @@ func (self *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (pa
 }
 
 // Cancel is called when the build is cancelled
-func (self *Builder) Cancel() {
-	if self.runner != nil {
```
Comment (Jalle19) [inline builder/upcloud/builder.go:114]: You sure this check is no longer needed? IIRC there was some reason for adding it.

File: builder/upcloud/builder.go [hunk 3]
```diff
@@ -20,59 +22,64 @@ type Builder struct {
...
+		return nil, nil, err
 	}
 
-	return nil, nil
+	return nil, nil, nil
 }
 
 // Run executes the actual build steps
-func (self *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packer.Artifact, error) {
+func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (packer.Artifact, error) {
```
Comment (Jalle19) [inline builder/upcloud/builder.go:61]: Same here, why change `self` to `b`?

File: builder/upcloud/builder.go (hunk 2 above)
Comment (moredure) [reply builder/upcloud/builder.go:114]: Currently it is handled by Run via context

File: builder/upcloud/builder.go (hunk 3 above)
Comment (moredure) [reply builder/upcloud/builder.go:61]: Just IDE recommendation

File: builder/upcloud/builder.go (hunk 2 above)
Comment (moredure) [reply builder/upcloud/builder.go:114]: I will check for sure in the evening

File: builder/upcloud/builder.go (hunk 3 above)
Comment (moredure) [reply builder/upcloud/builder.go:61]: "self", "this" is not the best practices for Golang receivers anyway...

File: builder/upcloud/step_create_server.go (hunk 1 above)
Comment (moredure) [reply builder/upcloud/step_create_server.go:22]: To distinguish service package and service instance

File: builder/upcloud/builder.go (hunk 2 above)
Comment (moredure) [reply builder/upcloud/builder.go:114]: @Jalle19 Works like a charm. Interrupts handled gracefully. Feel free to merge

Comment (moredure) [reply builder/upcloud/builder.go:114]: Cancel even removed from packer.Builder interface

**Comment by moredure** [comment]
Feel free to merge!

**Comment by Jalle19** [comment]
@jgabor

**Comment by moredure** [comment]
@jgabor

**Comment by jgabor** [comment]
Thanks for the PR and ping!

I'm working on other things at UpCloud nowadays, but I've pinged @jpjarven, who's responsible for our open source projects and he should get back to you as soon as he has a chance. :)

**Comment by moredure** [comment]
@jgabor, Thanks!

**Comment by ajmyyra** [comment]
Hi there, thank you for the PR! I've now merged it and will see if I could give the project some additional love & testing.

**Comment by moredure** [comment]
@ajmyyra Super cool!

**Pull request merged**